### ISSUE TITLE
It should not be able to focus elements behind the listing modal using tab keys

### ISSUE THREAD

**Issue by Rafi993** [forem/forem#10660]
**Describe the bug**

It should not be possible to focus elements behind the modal using tab keys. This should be prevented by focus trap for better accessibility

**To Reproduce**

1. Go to https://dev.to/listings/events/learn-how-to-contribute-to-open-source-k8j
2. Click on the text area
3. Press tab key three times
4. Now focus goes to element behind the modal hidden by the overlay

**Expected behavior**

User should not be able to focus elements behind the modal. When user keeps pressing tab key the focus should go to first focusable element inside the modal

**Screenshots**
![focus](https://user-images.githubusercontent.com/8092120/95197836-da8e2f80-07f7-11eb-8ada-995163b47e75.png)

In the screen shot the link behind the modal is focused

**Desktop:**

- OS, version: Ubuntu 20.04 LTS
- Browser, version: Version 1.14.81 Chromium: 85.0.4183.102 (Official Build) (64-bit)

**Comment by Rafi993** [comment]
I think it is related to https://github.com/forem/forem/issues/9585

**Comment by rhymes** [comment]
I think this is a duplicate of https://github.com/forem/forem/issues/10610, can you add your coment over there? Thanks @Rafi993!
